src/LP/LP.jl

```diff
@@ -118,30 +153,35 @@ function write_sense(io::IO, model::Model)
     return
 end
 
-function write_objective(io::IO, model::Model)
+function write_objective(io::IO, model::Model, sanitized_names::Dict{MOI.VariableIndex, String})
     print(io, "obj: ")
     obj_func_type = MOI.get(model, MOI.ObjectiveFunctionType())
     obj_func = MOI.get(model, MOI.ObjectiveFunction{obj_func_type}())
-    write_function(io, model, obj_func)
+    write_function(io, model, obj_func, sanitized_names)
     println(io)
     return
 end
 
 function MOI.write_to_file(model::Model, io::IO)
     MathOptFormat.create_unique_names(model)
+    sanitized_names = Dict{MOI.VariableIndex, String}()
+    for v in MOI.get(model, MOI.ListOfVariableIndices())
+        sanitized_names[v] = sanitized_name(MOI.get(model, MOI.VariableName(), v))
```

Comment:
Here, we could handle sanitized names that create duplicates. look in `create_unique_names` for the logic. Or at the very least, leave a TODO with a comment in the code.

Reply:
I just did it in the last two commits.